x/tokenfactory/keeper/genesis.go

```diff
@@ -24,14 +24,26 @@ func (k Keeper) InitGenesis(ctx sdk.Context, genState types.GenesisState) {
 		if err != nil {
 			panic(err)
 		}
+
 		err = k.createDenomAfterValidation(ctx, creator, genDenom.GetDenom())
 		if err != nil {
 			panic(err)
 		}
+
 		err = k.setAuthorityMetadata(ctx, genDenom.GetDenom(), genDenom.GetAuthorityMetadata())
 		if err != nil {
 			panic(err)
 		}
+
+		if _, err := sdk.AccAddressFromBech32(genDenom.HookContractAddress); genDenom.HookContractAddress != "" && err != nil {
```

Comment:
i think it's a redundant check, cause we have the address validation in `setBeforeSendHook` method

Reply:
I don't think we need any invalid addresses in the state and this method prevents such case